apps/api/src/app/auth/services/auth.service.ts

```diff
@@ -46,14 +47,15 @@ export class AuthService {
     distinctId: string,
     origin?: SignUpOriginEnum
   ) {
-    let user = await this.userRepository.findByLoginProvider(profile.id, authProvider);
+    const email = normalizeEmail(profile.email);
+    let user = await this.userRepository.findByEmail(email);
```

Comment:
Two things here:
- we should normalize the email
- we should look for the user by the email
Otherwise, it will be possible to create two accounts, one with sign-up form and another with a GH OAuth.

This change requires running the migration (check the description).

Reply:
I wasn't fully convinced of the normalized email solution but right now is the simplest way to solve this problem. Hopefully we can improve the authentication system in the future.
Many thanks for this amazing find and work. 👏🏻